flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/factory/KubernetesJobManagerFactoryTest.java

```diff
@@ -459,4 +464,28 @@ public void testEmptyHadoopConfDirectory() throws IOException {
                                                                 .getHadoopConfConfigMapName(
                                                                         CLUSTER_ID))));
     }
+
+    @Test
+    public void testCreateJobManagerDeploymentWithPodTemplate() throws Exception {
+        final FlinkPod flinkPod =
+                KubernetesUtils.loadPodFromTemplateFile(
+                        flinkKubeClient,
+                        KubernetesTestUtils.getJobManagerPodTemplatePath(),
+                        KubernetesJobManagerParameters.JOB_MANAGER_MAIN_CONTAINER_NAME);
+
+        final KubernetesJobManagerSpecification kubernetesJobManagerSpecification =
+                KubernetesJobManagerFactory.buildKubernetesJobManagerSpecification(
+                        flinkPod, kubernetesJobManagerParameters);
+
+        final PodTemplateSpec resultPod =
+                kubernetesJobManagerSpecification.getDeployment().getSpec().getTemplate();
+        // Pod template has been applied successfully.
+        assertThat(resultPod.getSpec().getInitContainers().size(), is(1));
+        assertThat(resultPod.getSpec().getContainers().size(), is(2));
+        assertThat(
+                resultPod.getSpec().getContainers().stream()
+                        .map(Container::getName)
+                        .collect(Collectors.toList()),
+                containsInAnyOrder("flink-job-manager", "sidecar-log-collector"));
```

Comment:
I think we need to verify that other functions also take effect after going through all the decorators, like what we do in `KubernetesUtilsTest` (metadata, init container, volumes, etc.) and `InitJob/TaskManagerDecoratorWithPodTemplateTest` (labels, annotations, env, etc.).